crates/polars-plan/src/logical_plan/aexpr/schema.rs

```diff
@@ -111,12 +111,22 @@ impl AExpr {
                         let mut field =
                             arena.get(*expr).to_field(schema, Context::Default, arena)?;
                         float_type(&mut field);
+                        if field.dtype == DataType::Date {
+                            field.coerce(DataType::Datetime(TimeUnit::Milliseconds, None));
+                        } else {
+                            float_type(&mut field);
+                        }
```

Comment:
I'm not familiar with this part of the codebase at all - @reswqa fancy taking a look?

Reply:
This section ensures that calling `df.lazy().select(pl.col("date_column").mean()).schema` is correct.